fetch.bs

```diff
@@ -2172,6 +2236,132 @@ exploits pertain to those <a for=request>types</a>. Also, considering "<code>ima
+</pre>
+
+<p>The value of the <code>Sec-Token-Binding</code> <a for=/>header</a> is a
+<dfn export id=token-binding-message>Token Binding Message</dfn> as specified in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-https#section-2">section 2</a>
+of the Token Binding over HTTP spec [[!TOKBIND-HTTPS]].
+
+<h4 id=federated-token-binding>Federated Token Binding</h4>
+<p>In the scenario described above, servers bind tokens to be used by user agents
+with those servers.
+To enable existing Web Authorization protocols to use bound credentials,
+there is a need to support scenarios where bound credentials issued by one server
+(e.g., an identity provider) are presented by the user agent to a different server
+(e.g., a relying party).
+[[TOKBIND-PROTOCOL]] and [[TOKBIND-HTTPS]] describe support for Federated Token Binding.
+
+<p>A <dfn export id=referred-token-binding-id>referred token binding ID</dfn> is the
+<a for=/>token binding ID</a> for a server other than the server to which the <a for=/>request</a> is being sent.
+
+Script code from an origin can set a <a for=/>request</a>'s
+<a for=request>use-referred-token-binding flag</a> to ask the user agent to send
+its <a for=/>token binding ID</a> as a <a for=/>referred token binding ID</a>
+in the <a for=/>Token Binding Message</a> sent to a target server.
+
+Alternately, servers that redirect a user agent to a different server can use the
+<code>Include-Referred-Token-Binding-ID</code> <a for=response>header</a>.
+
+<pre>
+Include-Referred-Token-Binding-ID = %x74.72.75.65 ; "true", case-sensitive
+</pre>
+
+<p>The value of this <a for=response>header</a> is specified in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-https#section-5.3">section 5.3</a>
+of the Token Binding over HTTP spec [[TOKBIND-HTTPS]].
+
+By setting the <code>Include-Referred-Token-Binding-ID</code>
+<a for=response>header</a> to "<code>true</code>",
+the <a for=request>origin</a> that sends the
+redirect <a for=/>response</a> tells the user agent to disclose the
+<a for=/>token binding ID</a> used by the user agent with that
+<a for=request>origin</a> to the target<a for=request>origin</a> as a
+<a for=/>referred token binding ID</a> in the <a for=/>Token Binding Message</a> sent to the target server.
+
+
+<h4 id=computing-token-binding-header-value>Computing a Token Binding Header Value</h4>
+<p>If the user agent supports <a for=/>Token Binding</a>,
+build a <code>Sec-Token-Binding</code> <a for=/>header</a>
+for a <a for=/>request</a> <var>httpRequest</var> using the Token Binding key store by running
+these substeps:
+
+<ol>
+ <li><p>Initialize <var>tokenBindingMessage</var> to null.
+
+ <li><p>Let <var>providedTokenBindingId</var> be the result of looking up the
+ <a for=/>token binding ID</a> for the <var>httpRequest</var>'s
+ <a for=/>current url</a>.
+
+ <li><p>If <var>providedTokenBindingId</var> is not null, then add a
+ <code>provided_token_binding</code> to <var>tokenBindingMessage</var>.
```

Comment:
`|tokenBindingMessage|` is `null` here. It's not clear what you mean by adding `|provided_token_binding|` to it.